### ISSUE TITLE
Source tarball build looks for vendored dependencies in the wrong place

### ISSUE THREAD

**Issue by ivan** [cockroachdb/cockroach#15439]
2. Please describe the issue you observed:

- What did you do?

I followed the "Build from Source" instructions on https://www.cockroachlabs.com/docs/install-cockroachdb.html on Ubuntu 16.04.2

- What did you expect to see?

A working binary :-)

- What did you see instead?

```
# PATH=/usr/lib/go-1.8/bin:$PATH make
make -C src/github.com/cockroachdb/cockroach all
make[1]: Entering directory '/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach'
mkdir -p /home/at/tarball/cockroach-latest/native/x86_64-linux-gnu/jemalloc
cd /home/at/tarball/cockroach-latest/native/x86_64-linux-gnu/jemalloc && /home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach/c-deps/jemalloc.src/configure 
checking for xsltproc... false
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether compiler is cray... no
checking whether compiler supports -std=gnu11... yes
checking whether compiler supports -Wall... yes
checking whether compiler supports -Werror=declaration-after-statement... yes
checking whether compiler supports -Wshorten-64-to-32... no
checking whether compiler supports -Wsign-compare... yes
checking whether compiler supports -pipe... yes
checking whether compiler supports -g3... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... no
checking size of void *... 8
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking size of intmax_t... 8
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether pause instruction is compilable... yes
checking for ar... ar
checking malloc.h usability... yes
checking malloc.h presence... yes
checking for malloc.h... yes
checking whether malloc_usable_size definition can use const argument... no
checking for library containing log... -lm
checking whether __attribute__ syntax is compilable... yes
checking whether compiler supports -fvisibility=hidden... yes
checking whether compiler supports -Werror... yes
checking whether compiler supports -herror_on_warning... no
checking whether tls_model attribute is compilable... yes
checking whether compiler supports -Werror... yes
checking whether compiler supports -herror_on_warning... no
checking whether alloc_size attribute is compilable... yes
checking whether compiler supports -Werror... yes
checking whether compiler supports -herror_on_warning... no
checking whether format(gnu_printf, ...) attribute is compilable... yes
checking whether compiler supports -Werror... yes
checking whether compiler supports -herror_on_warning... no
checking whether format(printf, ...) attribute is compilable... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking for ranlib... ranlib
checking for ld... /usr/bin/ld
checking for autoconf... false
checking for memalign... yes
checking for valloc... yes
checking whether compiler supports -O3... yes
checking whether compiler supports -funroll-loops... yes
checking configured backtracing method... N/A
checking for sbrk... yes
checking whether utrace(2) is compilable... no
checking whether valgrind is compilable... no
checking whether a program using __builtin_unreachable is compilable... yes
checking whether a program using __builtin_ffsl is compilable... yes
checking LG_PAGE... 12
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for pthread_create in -lpthread... yes
checking whether pthread_atfork(3) is compilable... yes
checking for library containing clock_gettime... none required
checking whether clock_gettime(CLOCK_MONOTONIC_COARSE, ...) is compilable... yes
checking whether clock_gettime(CLOCK_MONOTONIC, ...) is compilable... yes
checking whether mach_absolute_time() is compilable... no
checking whether compiler supports -Werror... yes
checking whether syscall(2) is compilable... yes
checking for secure_getenv... yes
checking for issetugid... no
checking for _malloc_thread_cleanup... no
checking for _pthread_mutex_init_calloc_cb... no
checking for TLS... yes
checking whether C11 atomics is compilable... yes
checking whether atomic(9) is compilable... no
checking whether Darwin OSAtomic*() is compilable... no
checking whether madvise(2) is compilable... yes
checking whether madvise(..., MADV_FREE) is compilable... no
checking whether madvise(..., MADV_DONTNEED) is compilable... yes
checking whether madvise(..., MADV_[NO]HUGEPAGE) is compilable... yes
checking whether to force 32-bit __sync_{add,sub}_and_fetch()... no
checking whether to force 64-bit __sync_{add,sub}_and_fetch()... no
checking for __builtin_clz... yes
checking whether Darwin os_unfair_lock_*() is compilable... no
checking whether Darwin OSSpin*() is compilable... no
checking whether glibc malloc hook is compilable... yes
checking whether glibc memalign hook is compilable... yes
checking whether pthreads adaptive mutexes is compilable... yes
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating jemalloc.pc
config.status: creating doc/html.xsl
config.status: creating doc/manpages.xsl
config.status: creating doc/jemalloc.xml
config.status: creating include/jemalloc/jemalloc_macros.h
config.status: creating include/jemalloc/jemalloc_protos.h
config.status: creating include/jemalloc/jemalloc_typedefs.h
config.status: creating include/jemalloc/internal/jemalloc_internal.h
config.status: creating test/test.sh
config.status: creating test/include/test/jemalloc_test.h
config.status: creating config.stamp
config.status: creating bin/jemalloc-config
config.status: creating bin/jemalloc.sh
config.status: creating bin/jeprof
config.status: creating include/jemalloc/jemalloc_defs.h
config.status: creating include/jemalloc/internal/jemalloc_internal_defs.h
config.status: creating test/include/test/jemalloc_test_defs.h
config.status: executing include/jemalloc/internal/private_namespace.h commands
config.status: executing include/jemalloc/internal/private_unnamespace.h commands
config.status: executing include/jemalloc/internal/public_symbols.txt commands
config.status: executing include/jemalloc/internal/public_namespace.h commands
config.status: executing include/jemalloc/internal/public_unnamespace.h commands
config.status: executing include/jemalloc/internal/size_classes.h commands
config.status: executing include/jemalloc/jemalloc_protos_jet.h commands
config.status: executing include/jemalloc/jemalloc_rename.h commands
config.status: executing include/jemalloc/jemalloc_mangle.h commands
config.status: executing include/jemalloc/jemalloc_mangle_jet.h commands
config.status: executing include/jemalloc/jemalloc.h commands
===============================================================================
jemalloc version   : 4.5.0-0-g04380e79f1e2428bd0ad000bbc6e3d2dfc6b66a5
library revision   : 2

CONFIG             : 
CC                 : gcc
CONFIGURE_CFLAGS   : -std=gnu11 -Wall -Werror=declaration-after-statement -Wsign-compare -pipe -g3 -fvisibility=hidden -O3 -funroll-loops
SPECIFIED_CFLAGS   : 
EXTRA_CFLAGS       : 
CPPFLAGS           : -D_GNU_SOURCE -D_REENTRANT
LDFLAGS            : 
EXTRA_LDFLAGS      : 
LIBS               : -lm  -lpthread
RPATH_EXTRA        : 

XSLTPROC           : false
XSLROOT            : 

PREFIX             : /usr/local
BINDIR             : /usr/local/bin
DATADIR            : /usr/local/share
INCLUDEDIR         : /usr/local/include
LIBDIR             : /usr/local/lib
MANDIR             : /usr/local/share/man

srcroot            : /home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach/c-deps/jemalloc.src/
abs_srcroot        : /home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach/c-deps/jemalloc.src/
objroot            : 
abs_objroot        : /home/at/tarball/cockroach-latest/native/x86_64-linux-gnu/jemalloc/

JEMALLOC_PREFIX    : 
JEMALLOC_PRIVATE_NAMESPACE
                   : je_
install_suffix     : 
malloc_conf        : 
autogen            : 0
cc-silence         : 1
debug              : 0
code-coverage      : 0
stats              : 1
prof               : 0
prof-libunwind     : 0
prof-libgcc        : 0
prof-gcc           : 0
tcache             : 1
thp                : 1
fill               : 1
utrace             : 0
valgrind           : 0
xmalloc            : 0
munmap             : 0
lazy_lock          : 0
tls                : 1
cache-oblivious    : 1
===============================================================================
can't load package: package github.com/Masterminds/glide: cannot find package "github.com/Masterminds/glide" in any of:
	/usr/lib/go-1.8/src/github.com/Masterminds/glide (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/Masterminds/glide (from $GOPATH)
can't load package: package github.com/client9/misspell/cmd/misspell: cannot find package "github.com/client9/misspell/cmd/misspell" in any of:
	/usr/lib/go-1.8/src/github.com/client9/misspell/cmd/misspell (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/client9/misspell/cmd/misspell (from $GOPATH)
can't load package: package github.com/cockroachdb/crlfmt: cannot find package "github.com/cockroachdb/crlfmt" in any of:
	/usr/lib/go-1.8/src/github.com/cockroachdb/crlfmt (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/crlfmt (from $GOPATH)
can't load package: package github.com/cockroachdb/stress: cannot find package "github.com/cockroachdb/stress" in any of:
	/usr/lib/go-1.8/src/github.com/cockroachdb/stress (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/stress (from $GOPATH)
can't load package: package github.com/golang/lint/golint: cannot find package "github.com/golang/lint/golint" in any of:
	/usr/lib/go-1.8/src/github.com/golang/lint/golint (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/golang/lint/golint (from $GOPATH)
can't load package: package github.com/google/pprof: cannot find package "github.com/google/pprof" in any of:
	/usr/lib/go-1.8/src/github.com/google/pprof (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/google/pprof (from $GOPATH)
can't load package: package github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway: cannot find package "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway" in any of:
	/usr/lib/go-1.8/src/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway (from $GOPATH)
can't load package: package github.com/jteeuwen/go-bindata/go-bindata: cannot find package "github.com/jteeuwen/go-bindata/go-bindata" in any of:
	/usr/lib/go-1.8/src/github.com/jteeuwen/go-bindata/go-bindata (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/jteeuwen/go-bindata/go-bindata (from $GOPATH)
can't load package: package github.com/kisielk/errcheck: cannot find package "github.com/kisielk/errcheck" in any of:
	/usr/lib/go-1.8/src/github.com/kisielk/errcheck (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/kisielk/errcheck (from $GOPATH)
can't load package: package github.com/mattn/goveralls: cannot find package "github.com/mattn/goveralls" in any of:
	/usr/lib/go-1.8/src/github.com/mattn/goveralls (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/mattn/goveralls (from $GOPATH)
can't load package: package github.com/mdempsky/unconvert: cannot find package "github.com/mdempsky/unconvert" in any of:
	/usr/lib/go-1.8/src/github.com/mdempsky/unconvert (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/mdempsky/unconvert (from $GOPATH)
can't load package: package github.com/mibk/dupl: cannot find package "github.com/mibk/dupl" in any of:
	/usr/lib/go-1.8/src/github.com/mibk/dupl (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/mibk/dupl (from $GOPATH)
can't load package: package github.com/wadey/gocovmerge: cannot find package "github.com/wadey/gocovmerge" in any of:
	/usr/lib/go-1.8/src/github.com/wadey/gocovmerge (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/wadey/gocovmerge (from $GOPATH)
can't load package: package golang.org/x/tools/cmd/goimports: cannot find package "golang.org/x/tools/cmd/goimports" in any of:
	/usr/lib/go-1.8/src/golang.org/x/tools/cmd/goimports (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/golang.org/x/tools/cmd/goimports (from $GOPATH)
can't load package: package golang.org/x/tools/cmd/goyacc: cannot find package "golang.org/x/tools/cmd/goyacc" in any of:
	/usr/lib/go-1.8/src/golang.org/x/tools/cmd/goyacc (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/golang.org/x/tools/cmd/goyacc (from $GOPATH)
can't load package: package golang.org/x/tools/cmd/guru: cannot find package "golang.org/x/tools/cmd/guru" in any of:
	/usr/lib/go-1.8/src/golang.org/x/tools/cmd/guru (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/golang.org/x/tools/cmd/guru (from $GOPATH)
can't load package: package golang.org/x/tools/cmd/stringer: cannot find package "golang.org/x/tools/cmd/stringer" in any of:
	/usr/lib/go-1.8/src/golang.org/x/tools/cmd/stringer (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/golang.org/x/tools/cmd/stringer (from $GOPATH)
build/common.mk:170: recipe for target '/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach/bin/.bootstrap' failed
make[1]: *** [/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach/bin/.bootstrap] Error 1
make[1]: Leaving directory '/home/at/tarball/cockroach-latest/src/github.com/cockroachdb/cockroach'
Makefile:4: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by sunhay** [comment]
Cloning and building failed for me as well with a similar issue. 

I ended up having to do the following: 
```
$ go get -d -v github.com/cockroachdb/cockroach
$ cd $GOPATH/src/github.com/cockroachdb
$ make build
```

**Comment by benesch** [comment]
@ivan, sorry you're running into this trouble! I'm afraid I can't seem to reproduce your issue. Could you post the output of `go env` on your machine? The output of `env` would be helpful too, if you wouldn't mind to strip any sensitive information you might have set in your environment.

@sunhay, if you clone the repository with Git (i.e., a `git clone` instead of a `go get`), you'll need to make sure you're cloning into a valid `$GOPATH`:

```bash
export GOPATH=~/foo
mkdir $GOPATH/src/github.com/cockroachdb
git clone git@github.com:cockroachdb/cockroach.git $GOPATH/src/github.com/cockroachdb/cockroach
```

`go get` makes this slightly more convenient, of course, by refusing to install anywhere _but_ $GOPATH/src/github.com/cockroachdb/cockroach.

Our source tarballs are, modulo the bug that @ivan is experiencing, the only supported method of building outside of your `GOPATH`. And, in fact, we accomplish this by arranging the tarball contents such that they form a valid `GOPATH` when extracted and ignoring any `GOPATH` you have set in your environment.

**Comment by sunhay** [comment]
Got it! Thanks for the info @benesch

**Comment by ivan** [comment]
```
at@sandlin:/home/at/cockroach-latest# PATH=/usr/lib/go-1.8/bin:$PATH go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/at/go"
GORACE=""
GOROOT="/usr/lib/go-1.8"
GOTOOLDIR="/usr/lib/go-1.8/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build912604670=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

```
at@sandlin:/home/at/cockroach-latest# PATH=/usr/lib/go-1.8/bin:$PATH env   
LANG=en_US.UTF-8
USER=at
LOGNAME=at
HOME=/home/at
PATH=/usr/lib/go-1.8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
MAIL=/var/mail/at
SHELL=/bin/zsh
SSH_CLIENT=[...]
SSH_CONNECTION=[...]
SSH_TTY=/dev/pts/0
TERM=xterm
XDG_SESSION_ID=2
XDG_RUNTIME_DIR=/run/user/1000
SHLVL=1
PWD=/home/at/cockroach-latest
OLDPWD=/home/at
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
EDITOR=nano
PAGER=less
LESS=-IMx4R
LESSHISTFILE=-
_=/usr/bin/env
```

```
# dpkg -l | grep golang
ii  golang-1.8                           1.8.1-1                         all          Go programming language compiler - metapackage
ii  golang-1.8-doc                       1.8.1-1                         all          Go programming language - documentation
ii  golang-1.8-go                        1.8.1-1                         amd64        Go programming language compiler, linker, compiled stdlib
ii  golang-1.8-src                       1.8.1-1                         amd64        Go programming language - source files
```

**Comment by xphoniex** [comment]
@ivan can you confirm that you have a `/vendor` folder with these packages?

```
can't load package: package github.com/Masterminds/glide: cannot find package "github.com/Masterminds/glide" in any of:
	/usr/lib/go-1.8/src/github.com/Masterminds/glide (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/Masterminds/glide (from $GOPATH)
can't load package: package github.com/client9/misspell/cmd/misspell: cannot find package "github.com/client9/misspell/cmd/misspell" in any of:
	/usr/lib/go-1.8/src/github.com/client9/misspell/cmd/misspell (from $GOROOT)
	/home/at/tarball/cockroach-latest/src/github.com/client9/misspell/cmd/misspell (from $GOPATH)
.
.
.
```

**Comment by ivan** [comment]
```
at@sandlin:/home/at/tarball/cockroach-latest# ls src/github.com/cockroachdb/cockroach/vendor/github.com/Masterminds/glide
.gitignore   CHANGELOG.md  Makefile   action/       cache/  dependency/  gb/       glide.lock  glide_test.go  gom/  importer/  mkdocs.yml  path/  testdata/  util/
.travis.yml  LICENSE       README.md  appveyor.yml  cfg/    docs/        glide.go  glide.yaml  godep/         gpm/  mirrors/   msg/        repo/  tree/      vendor/

at@sandlin:/home/at/tarball/cockroach-latest# ls src/github.com/cockroachdb/cockroach/vendor/github.com/client9/misspell/cmd/misspell 
main.go
```

**Comment by benesch** [comment]
@ivan, I just managed to reproduce. Thanks for your patience! Working on a fix.

**Comment by benesch** [comment]
Looks like it was a bug in our Makefile's `GOPATH` override. Sorry about that, @ivan, and thanks so much for reporting!

The following patch fixes it for me:

```patch
--- a/src/github.com/cockroachdb/cockroach/Makefile
+++ b/src/github.com/cockroachdb/cockroach/Makefile
@@ -110,7 +110,7 @@ short: build testshort checkshort
 
 buildoss: BUILDTARGET = ./pkg/cmd/cockroach-oss
 
-build buildoss: BUILDMODE = build -i -o cockroach$(SUFFIX)$(shell $(XGO) env GOEXE)
+build buildoss: BUILDMODE = build -i -o cockroach$(SUFFIX)$$($(XGO) env GOEXE)
 
 # The build.utcTime format must remain in sync with TimeFormat in pkg/build/info.go.
 build buildoss install: override LINKFLAGS += \
diff --git a/build/common.mk b/build/common.mk
index 009290b2e..be12c628c 100644
--- a/src/github.com/cockroachdb/cockroach/build/common.mk
+++ b/src/github.com/cockroachdb/cockroach/build/common.mk
@@ -170,7 +170,7 @@ ifneq ($(GIT_DIR),)
 	git submodule update --init
 endif
 	@$(GO_INSTALL) -v $(PKG_ROOT)/cmd/{metacheck,ncpus,returncheck} \
-	$(shell $(GO) list -tags glide -f '{{join .Imports "\n"}}' $(REPO_ROOT)/build)
+	$$($(GO) list -tags glide -f '{{join .Imports "\n"}}' $(REPO_ROOT)/build)
 	touch $@
 
 # Make doesn't expose a list of the variables declared in a given file, so we
```

I'll work on getting that fix into our next release's source tarball. For now, though, you can apply the patch manually to the 1.0-rc.1 source tarball. Full steps:

```bash
curl -fsSL https://binaries.cockroachdb.com/cockroach-latest.src.tgz | tar xz
cd cockroach-latest
curl https://gist.githubusercontent.com/benesch/fc3091aa4ca5ea9c7effcbc41dbd1c2b/raw/80ad604f770841071ab34b88cab7b208a0547ec5/crdb-1.0-rc.1-build-from-source.patch | patch -p1
make build
```

Could you verify the above works for you when you have a moment?

**Comment by benesch** [comment]
This is merged and will ship in 1.0-rc.2 later this week. @ivan, please give the patch a shot in the meantime and reopen this issue if your build is still broken.
